tests/main/test-user-daemon-in-graphical-environment/task.yaml

```
@@ -0,0 +1,40 @@
+summary: Ensure that the graphical user daemons work as expected
+
+details: |
+    This test ensures that an user daemon with the desktop, x11 or wayland plugs
+    will be launched with the destkop session only.
+
+# Same systems than `interfaces-desktop`, since we require that plug.
+systems: [-ubuntu-core-*, -ubuntu-*-ppc64el, -fedora-*, -opensuse-*, -amazon-*, -centos-*]
+
+prepare: |
+    snap set system experimental.user-daemons=true
+    "$TESTSTOOLS"/snaps-state install-local test-implicit-graphical-user-daemon
+
+execute: |
+    NORMAL_SERVICE=/etc/systemd/user/snap.test-implicit-graphical-user-daemon.normal.service
+    DESKTOP_SERVICE=/etc/systemd/user/snap.test-implicit-graphical-user-daemon.graphical.service
+    DESKTOP_DBUS_SERVICE1=/etc/systemd/user/snap.test-implicit-graphical-user-daemon.graphical-dbus.service
+    DESKTOP_DBUS_SERVICE2=/etc/systemd/user/snap.test-implicit-graphical-user-daemon.graphical-dbus2.service
+
+    echo "Ensure graphical daemon has the right dependencies"
+
+    test -f ${NORMAL_SERVICE}
+    test -f ${DESKTOP_SERVICE}
+    test -f ${DESKTOP_DBUS_SERVICE1}
+    test -f ${DESKTOP_DBUS_SERVICE2}
+
+    grep "WantedBy=graphical-session\.target" ${DESKTOP_SERVICE}
+    grep "After=graphical-session\.target" ${DESKTOP_SERVICE}
+    grep "Requisite=graphical-session\.target" ${DESKTOP_SERVICE}
+    grep "BindsTo=graphical-session\.target" ${DESKTOP_SERVICE}
```

Generally we want to use `MATCH` instead of grep, as it provides some more context.

```suggestion
    MATCH "WantedBy=graphical-session\.target" < "${DESKTOP_SERVICE}"
    MATCH "After=graphical-session\.target" < "${DESKTOP_SERVICE}"
    MATCH "Requisite=graphical-session\.target" < "${DESKTOP_SERVICE}"
    MATCH "BindsTo=graphical-session\.target" < "${DESKTOP_SERVICE}"
```